Python/ast.c

```diff
@@ -21,6 +21,25 @@ static int validate_nonempty_seq(asdl_seq *, const char *, const char *);
 static int validate_stmt(stmt_ty);
 static int validate_expr(expr_ty, expr_context_ty);
 
+static int
+validate_name(PyObject *name)
+{
+    assert(PyUnicode_Check(name));
+    static const char * const forbidden[] = {
```

Comment:
We already have `FORBIDDEN`

Reply:
I am aware of that but usage of `__debug__` is OK, since it's constant form is a special compiler optimization and people shouldn't be aware of it. The current `FORBIDDEN` lists names only for store context, but this is for load context too.

Reply:
I don't fully understand what you are saying but IIRC `compiler_nameop` checks for `__debug__` as well

Reply:
What I am saying is that `ast.Name(__debug__, ast.Load())` is totally OK, since it is going to be converted to constant in ast optimizer as an optimization, and people shouldn't have to know it is a constant.

Reply:
https://github.com/python/cpython/blob/087d612efebe7c64e5f079b07e0454111859830e/Python/ast_opt.c#L565-L570

Reply:
> What I am saying is that ast.Name(__debug__, ast.Load()) is totally OK,

You mean `ast.Name("__debug__", ast.Load())`, no?

I understand that, but I thjink it makes it inconsistent with the behaviour of `Load` context. What I am saying is that all names that will fail in an assignment should fail here as well.

Reply:
> You mean ast.Name("__debug__", ast.Load()), no?

Yes

> I understand that, but I think it makes it inconsistent with the behaviour of Load context. What I am saying is that all names that will fail in an assignment should fail here as well.

The purpose of this warning is to prevent the compilation of the following AST: `ast.Name("True"/"False"/"None", any_context)`. The `ast.Name("__debug__", ast.Load())` is the only way someone can construct an AST and that loads `__debug__`. There is a replacement for this in the `ast.Constant` form. What I was trying to explain is that this becomes an `ast.Constant` inside of the ast optimizer, and it is an optimization.